File: src/test/programmers/level2/ExpressionOfNumber.java
package test.programmers.level2;

public class ExpressionOfNumber {
    public static void main(String[] args) {
        System.out.println(solution(15));
    }

    static int solution(int n) {
        int answer = 0;
        int temp;

        for(int i = 1; i <= n; i++) {
            temp = 0;

            for(int j = i; j <= n; j++) {
                temp += j;

                if(temp == n) {
                    answer++;
                    break;
                } else if(temp > n){
                    break;
                }
            }
        }

        return answer;
    }
}

File: src/test/programmers/python/level2/ExpressionOfNumber.py
def solution(n):
    answer = 1
    limit = n // 2 + 1
    for i in range(1, limit):
        _sum, num = 0, i
        while _sum <= n:
            _sum += num
            num += 1
            if _sum == n:
                answer += 1
                break

    return answer

print(solution(15))